manager/dispatcher/dispatcher_test.go

```diff
@@ -1290,54 +1472,150 @@ func filterDependencies(secrets []*api.Secret, configs []*api.Config, inTasks, n
 	var (
-		secrets []*api.Secret
-		configs []*api.Config
-		tasks   []*api.Task
+		secrets      []*api.Secret
+		configs      []*api.Config
+		resourceRefs []*api.ResourceReference
+		tasks        []*api.Task
 	)
 	for i := 0; i <= len(taskStatesInOrder); i++ {
-		secrets = append(secrets, &api.Secret{
-			ID: fmt.Sprintf("IDsecret%d", i),
-			Spec: api.SecretSpec{
-				Annotations: api.Annotations{
-					Name: fmt.Sprintf("secret%d", i),
-				},
-				Data: []byte(fmt.Sprintf("secret%d", i)),
-			},
-		})
-		configs = append(configs, &api.Config{
-			ID: fmt.Sprintf("IDconfig%d", i),
-			Spec: api.ConfigSpec{
-				Annotations: api.Annotations{
-					Name: fmt.Sprintf("config%d", i),
-				},
-				Data: []byte(fmt.Sprintf("config%d", i)),
-			},
+		secrets = append(secrets, mockNumberedSecret(i))
+		configs = append(configs, mockNumberedConfig(i))
+
+		resourceRefs = append(resourceRefs, &api.ResourceReference{
+			ResourceID:   fmt.Sprintf("IDresourceRefSecret%d", i),
+			ResourceType: api.ResourceType_SECRET,
+		}, &api.ResourceReference{
+			ResourceID:   fmt.Sprintf("IDresourceRefConfig%d", i),
+			ResourceType: api.ResourceType_CONFIG,
 		})
 	}
+
+	for i, taskState := range taskStatesInOrder {
+		spec := taskSpecFromDependencies(secrets[i], secrets[len(secrets)-1], configs[i], configs[len(configs)-1], resourceRefs[2*i], resourceRefs[2*i+1])
+		tasks = append(tasks, mockNumberedReadyTask(i, nodeID, taskState, spec))
+	}
+	return secrets, configs, resourceRefs, tasks
+}
+
+// Creates 1 task for every possible task state, so there are 12 tasks, ID=0-11 inclusive.
+// Creates 1 secret and 1 config for every single task state + 1, so there are 13 secrets, 13 configs, ID=0-12 inclusive
+// For each task, the dependencies assigned to it are: secret<i>, secret12, config<i>, config12.
+// There are no ResourceReferences in these TaskSpecs
+func makeTasksAndDependenciesNoResourceReferences(t *testing.T, nodeID string) ([]*api.Secret, []*api.Config, []*api.ResourceReference, []*api.Task) {
+	var (
+		secrets      []*api.Secret
+		configs      []*api.Config
+		resourceRefs []*api.ResourceReference
+		tasks        []*api.Task
+	)
+	for i := 0; i <= len(taskStatesInOrder); i++ {
+		secrets = append(secrets, mockNumberedSecret(i))
+		configs = append(configs, mockNumberedConfig(i))
+	}
 	for i, taskState := range taskStatesInOrder {
 		spec := taskSpecFromDependencies(secrets[i], secrets[len(secrets)-1], configs[i], configs[len(configs)-1])
-		tasks = append(tasks, &api.Task{
-			NodeID:       nodeID,
-			ID:           fmt.Sprintf("testTask%d", i),
-			Status:       api.TaskStatus{State: taskState},
-			DesiredState: api.TaskStateReady,
-			Spec:         spec,
+		tasks = append(tasks, mockNumberedReadyTask(i, nodeID, taskState, spec))
+	}
+	return secrets, configs, resourceRefs, tasks
+}
+
+// Creates 1 task for every possible task state, so there are 12 tasks, ID=0-11 inclusive.
+// Creates 1 secret and 1 config per task by resource reference
```

Comment:
This seems to create 1 secret and 1 config per task, plus 1 extra secret and 1 extra config.
However, it does not assign that last secret and config to any tests, so for each task, the dependencies assigned to it are actually just `resourceRefSecret<i>` and `<resourceRefConfig<i>`